
# Проверка гипотез с помощью python.

## Задание

* Выполненить проверку предложенной гипотезы
* Выполненить проверку статистической гипотезы для проверки нормальности числовых признаков
* Проверка мультиколлинеарности
* Все статистические тесты должны быть выполнены с 95%-ным уровнем достоверности (т.е. значение р < 0,05)

## Датасет

Как данные представлены заказчиком:

<table>
<thead><tr>
<th><strong>Feature Name</strong></th>
<th><strong>Description</strong></th>
<th><strong>Data Type</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>customerID</td>
<td>Содержит идентификатор клиента</td>
<td>categorical</td>
</tr>
<tr>
<td>gender</td>
<td>Пол клиента</td>
<td>categorical</td>
</tr>
<tr>
<td>SeniorCitizen</td>
<td>Является ли клиент пожилым гражданином</td>
<td>numeric, int</td>
</tr>
<tr>
<td>Partner</td>
<td>Есть ли у клиента партнер</td>
<td>categorical</td>
</tr>
<tr>
<td>Dependents</td>
<td>Является ли клиент кормильцем</td>
<td>categorical</td>
</tr>
<tr>
<td>tenure</td>
<td>Количество месяцев, в течение которых клиент оставался в компании</td>
<td>numeric, int</td>
</tr>
<tr>
<td>PhoneService</td>
<td>Есть ли у клиента телефонная связь</td>
<td>categorical</td>
</tr>
<tr>
<td>MultipleLines</td>
<td>Есть ли у клиента несколько линий</td>
<td>categorical</td>
</tr>
<tr>
<td>InternetService</td>
<td>Интернет-провайдер клиента</td>
<td>categorical</td>
</tr>
<tr>
<td>OnlineSecurity</td>
<td>Есть ли у клиента онлайн-безопасность</td>
<td>categorical</td>
</tr>
<tr>
<td>OnlineBackup</td>
<td>Есть ли у клиента онлайн-резервное копирование</td>
<td>categorical</td>
</tr>
<tr>
<td>DeviceProtection</td>
<td>Имеет ли клиент защиту устройства</td>
<td>categorical</td>
</tr>
<tr>
<td>TechSupport</td>
<td>Есть ли у клиента техническая поддержка</td>
<td>categorical</td>
</tr>
<tr>
<td>streamingTV</td>
<td>Есть ли у клиента потоковое телевидение</td>
<td>categorical</td>
</tr>
<tr>
<td>streamingMovies</td>
<td>Есть ли у клиента стриминговые сервисы</td>
<td>categorical</td>
</tr>
<tr>
<td>Contract</td>
<td>Срок действия контракта заказчика</td>
<td>categorical</td>
</tr>
<tr>
<td>PaperlessBilling</td>
<td>имеет ли клиент безналичные счета</td>
<td>categorical</td>
</tr>
<tr>
<td>PaymentMethod</td>
<td>Способ оплаты клиента</td>
<td>categorical</td>
</tr>
<tr>
<td>MonthlyCharges</td>
<td>Сумма, взимаемая с клиента ежемесячно </td>
<td> numeric , int</td>
</tr>
<tr>
<td>TotalCharges</td>
<td>Общая сумма, списанная с клиента</td>
<td>object</td>
</tr>
<tr>
<td>Churn</td>
<td>Ушел ли клиент</td>
<td>categorical</td>
</tr>
</tbody>
</table>

Описание от заказчика:

Каждая строка представляет клиента, каждый столбец содержит атрибуты клиента.

Набор данных включает информацию о:

- Клиенты, которые ушли в течение последнего месяца - колонка называется `Churn`
- Услуги, на которые подписался каждый клиент - телефон, несколько линий, интернет, онлайн-безопасность, резервное копирование, защита устройств, техническая поддержка, потоковое ТВ и фильмы.
- Информация о счетах клиентов - как долго они являются клиентами, контракт, способ оплаты, безналичные счета, ежемесячные платежи и общая сумма платежей.
- Демографическая информация о клиентах - пол, возраст, наличие партнеров и иждивенцев.

## 1. Импорт Библиотек 

Импортируйте необходимые библиотеки

In [45]:
# %pip install scipy

In [46]:
import numpy as np
import pandas as pd
pd.set_option('display.max_columns', None) # отобразить все колонки

from scipy import stats

## 2. Данные

In [47]:
df = pd.read_csv("./churn.csv")
df.head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity OnlineBackup  \
0  No phone service             DSL             No          Yes   
1                No             DSL            Yes           No   
2                No             DSL            Yes          Yes   
3  No phone service             DSL            Yes           No   
4                No     Fiber optic             No           No   

  DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0               No          No          No              No  Month-to-month   
1              Yes          No          No              No        One year   
2               No          No          No              No  Month-to-month   
3              Yes         Yes          No              No        One year   
4               No          No          No              No  Month-to-month   

  PaperlessBilling              PaymentMethod  MonthlyCharges TotalCharges  \
0              Yes           Electronic check           29.85        29.85   
1               No               Mailed check           56.95       1889.5   
2              Yes               Mailed check           53.85       108.15   
3               No  Bank transfer (automatic)           42.30      1840.75   
4              Yes           Electronic check           70.70       151.65   

  Churn  
0    No  
1    No  
2   Yes  
3    No  
4   Yes

## 2. Описательный анализ данных

Рассмотрите данные - изучите медианы, средние, дисперсии и т.д. 

Что можете сказать о данных? Сделайте вывод.

In [48]:
# ваше решение
statistics = pd.DataFrame({
    'Среднее': df.loc[:,['SeniorCitizen','tenure','MonthlyCharges']].mean(),
    'Медиана': df.loc[:,['SeniorCitizen','tenure','MonthlyCharges']].median(),
    'Дисперсия': df.loc[:,['SeniorCitizen','tenure','MonthlyCharges']].var(),
    'Стандартное отклонение': df.loc[:,['SeniorCitizen','tenure','MonthlyCharges']].std()
})

statistics.T

SeniorCitizen      tenure  MonthlyCharges
Среднее                      0.162147   32.371149       64.761692
Медиана                      0.000000   29.000000       70.350000
Дисперсия                    0.135875  603.168108      905.410934
Стандартное отклонение       0.368612   24.559481       30.090047

**Вывод:**
- Большинство клиентов не являются пожилыми (среднее значение SeniorCitizen 0.162147).
- Дисперсия столбца SeniorCitizen 0.135875, что означает, что значения в этом столбце распределены относительно равномерно от среднего значения.
- Стандартное отклонение 0.368612 также показывает разброс данных в столбце.

- Средняя продолжительность нахождения клиента в компании около 32 месяцев.
- Медиана 29.000000 указывает на то, что половина клиентов провели в компании менее 29 месяцев.
- Высокая дисперсия (603.168108) и стандартное отклонение (24.559481) в столбце Tenure могут указывать на значительный разброс в данных.

- Клиенты в среднем тратят 64.761692 на услуги каждый месяц.
- Медиана 70.350000 указывает на то, что половина клиентов платит более 70.35 в месяц за услуги.
- Столбец имеет довольно высокую дисперсию (905.410934) и стандартное отклонение (30.090047), что указывает на значительный разброс данных.

### Уникальные значения для всех столбцов

Рассмотрите уникальные значения категориальных признаков. 

Сделайте вывод.

**Подсказка:** можете воспользоваться методом `describe(include='object')`

In [49]:
# ваше решение
df.describe(include='object').T

count unique               top  freq
customerID        7043   7043        7590-VHVEG     1
gender            7043      2              Male  3555
Partner           7043      2                No  3641
Dependents        7043      2                No  4933
PhoneService      7043      2               Yes  6361
MultipleLines     7043      3                No  3390
InternetService   7043      3       Fiber optic  3096
OnlineSecurity    7043      3                No  3498
OnlineBackup      7043      3                No  3088
DeviceProtection  7043      3                No  3095
TechSupport       7043      3                No  3473
StreamingTV       7043      3                No  2810
StreamingMovies   7043      3                No  2785
Contract          7043      3    Month-to-month  3875
PaperlessBilling  7043      2               Yes  4171
PaymentMethod     7043      4  Electronic check  2365
TotalCharges      7043   6531                      11
Churn             7043      2                No  5174

**Вывод:**
- Всего в таблице содержится 7043 записи.
- Уникальных значений в категориальных столбцах в среднем 2-3, в PaymentMethod - 4.
- Телефонная связь есть у большинства клиентов (6361 раз), другие услуги клиенты чаще не используют.
- Партнеры есть у половины клиентов (3641), но большинство являются кормильцами (4933).
- Большинство клиентов используют услугу Fiber optic в столбце InternetService (3096).
- Большинство клиентов обновляют договор ежемесячно (3875).
- Большинство клиентов выбирают электронный чек в качестве оплаты (2365).
- В основном клиенты остаются (5174).

## 3. Предобработка данных

### Дубликаты

Предобработайте датасет - проверьте на наличие дубликатов и удалите, если они есть.

In [50]:
# ваше решение
df.duplicated().sum()

np.int64(0)

**Вывод:**
Пропусков нет

### Отсутствующие значения

Пояснение к данному датасету:

Не всегда в данных есть явные пропуски, которые легко индексируются средствами Pandas. К сожалению бывают проблемы с выгрузками данных, некачественными источниками, или когда данные записываются вручную - часто присутствует человеческий фактов. По мере работы с новыми данными лучше проверять на наличие таких данных, иначе из-за этого исследование может исказиться.

Рассмотрим такой случай:

In [51]:
df.isna().sum()

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

Да, не видно пропусков. Но почему тогда столбец `TotalCharges` не отобразился изначально как числовой столбец?

Проверим все столбцы:

In [52]:
for col in df:
    if df[col].dtype == 'object':
        nans = df[col].apply(lambda x: len(x.strip())==0).sum()
        if nans > 0:
            print(f'Неявные пропуски столбца {col}:', df[col].apply(lambda x: len(x.strip())==0).sum())

Неявные пропуски столбца TotalCharges: 11


Что мы можем сделать в таком случае?

1) По этому же фильтру удалить строки с пропусками, чтобы не искажать данные
2) Перевести тип столбца на числовой и с параметром `errors = 'coerce'` и удалить

Но можно не удалять, чтобы потерять данные, а:

1) Заменить значения на среднюю
2) Поменять на другое - но только в том случае, если есть какая-то зависимость в данных.

Давайте изменим тип и найдем зависимость:

In [53]:
df["TotalCharges"] = pd.to_numeric(df["TotalCharges"], errors = 'coerce')

df[df["TotalCharges"].isna()]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
488   4472-LVYGI  Female              0     Yes        Yes       0   
753   3115-CZMZD    Male              0      No        Yes       0   
936   5709-LVOEQ  Female              0     Yes        Yes       0   
1082  4367-NUYAO    Male              0     Yes        Yes       0   
1340  1371-DWPAZ  Female              0     Yes        Yes       0   
3331  7644-OMVMY    Male              0     Yes        Yes       0   
3826  3213-VVOLG    Male              0     Yes        Yes       0   
4380  2520-SGTTA  Female              0     Yes        Yes       0   
5218  2923-ARZLG    Male              0     Yes        Yes       0   
6670  4075-WKNIU  Female              0     Yes        Yes       0   
6754  2775-SEFEE    Male              0      No        Yes       0   

     PhoneService     MultipleLines InternetService       OnlineSecurity  \
488            No  No phone service             DSL                  Yes   
753           Yes                No              No  No internet service   
936           Yes                No             DSL                  Yes   
1082          Yes               Yes              No  No internet service   
1340           No  No phone service             DSL                  Yes   
3331          Yes                No              No  No internet service   
3826          Yes               Yes              No  No internet service   
4380          Yes                No              No  No internet service   
5218          Yes                No              No  No internet service   
6670          Yes               Yes             DSL                   No   
6754          Yes               Yes             DSL                  Yes   

             OnlineBackup     DeviceProtection          TechSupport  \
488                    No                  Yes                  Yes   
753   No internet service  No internet service  No internet service   
936                   Yes                  Yes                   No   
1082  No internet service  No internet service  No internet service   
1340                  Yes                  Yes                  Yes   
3331  No internet service  No internet service  No internet service   
3826  No internet service  No internet service  No internet service   
4380  No internet service  No internet service  No internet service   
5218  No internet service  No internet service  No internet service   
6670                  Yes                  Yes                  Yes   
6754                  Yes                   No                  Yes   

              StreamingTV      StreamingMovies  Contract PaperlessBilling  \
488                   Yes                   No  Two year              Yes   
753   No internet service  No internet service  Two year               No   
936                   Yes                  Yes  Two year               No   
1082  No internet service  No internet service  Two year               No   
1340                  Yes                   No  Two year               No   
3331  No internet service  No internet service  Two year               No   
3826  No internet service  No internet service  Two year               No   
4380  No internet service  No internet service  Two year               No   
5218  No internet service  No internet service  One year              Yes   
6670                  Yes                   No  Two year               No   
6754                   No                   No  Two year              Yes   

                  PaymentMethod  MonthlyCharges  TotalCharges Churn  
488   Bank transfer (automatic)           52.55           NaN    No  
753                Mailed check           20.25           NaN    No  
936                Mailed check           80.85           NaN    No  
1082               Mailed check           25.75           NaN    No  
1340    Credit card (automatic)           56.05           NaN    No  
3331               Mailed check           19.85       

**Но как выглядят наши данные обычно?**

In [54]:
df.sample(5)

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
4274  3572-UUHRS    Male              0      No         No      45   
2894  8869-TORSS  Female              0      No         No      48   
6101  4627-MIHJH  Female              1      No         No      69   
4265  2632-UCGVD    Male              1     Yes         No      66   
5907  0082-LDZUE    Male              0      No         No       1   

     PhoneService MultipleLines InternetService       OnlineSecurity  \
4274          Yes            No              No  No internet service   
2894          Yes            No             DSL                  Yes   
6101          Yes           Yes     Fiber optic                  Yes   
4265          Yes           Yes     Fiber optic                   No   
5907          Yes            No             DSL                   No   

             OnlineBackup     DeviceProtection          TechSupport  \
4274  No internet service  No internet service  No internet service   
2894                   No                  Yes                  Yes   
6101                   No                  Yes                  Yes   
4265                   No                   No                  Yes   
5907                   No                   No                   No   

              StreamingTV      StreamingMovies        Contract  \
4274  No internet service  No internet service        One year   
2894                   No                   No        One year   
6101                  Yes                  Yes        Two year   
4265                  Yes                  Yes  Month-to-month   
5907                   No                   No  Month-to-month   

     PaperlessBilling              PaymentMethod  MonthlyCharges  \
4274               No  Bank transfer (automatic)           18.85   
2894              Yes    Credit card (automatic)           60.35   
6101               No    Credit card (automatic)          110.25   
4265              Yes    Credit card (automatic)          100.05   
5907              Yes               Mailed check           44.30   

      TotalCharges Churn  
4274        867.30    No  
2894       2896.40   Yes  
6101       7467.55    No  
4265       6871.90   Yes  
5907         44.30    No

**Ничего не заметили?**

Может общая сумма, списанная с клиента, равна сумме, взимаемой с клиента ежемесячно умноженное на количество времени?

Давайте посмострим.

In [55]:
df[(df['TotalCharges'] / df['MonthlyCharges']) == df['tenure']].head()

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0   7590-VHVEG  Female              0     Yes         No       1           No   
20  8779-QRDMV    Male              1      No         No       1           No   
22  1066-JKSGK    Male              0      No         No       1          Yes   
27  8665-UTDHZ    Male              0     Yes        Yes       1           No   
33  7310-EGVHZ    Male              0      No         No       1          Yes   

       MultipleLines InternetService       OnlineSecurity  \
0   No phone service             DSL                   No   
20  No phone service             DSL                   No   
22                No              No  No internet service   
27  No phone service             DSL                   No   
33                No              No  No internet service   

           OnlineBackup     DeviceProtection          TechSupport  \
0                   Yes                   No                   No   
20                   No                  Yes                   No   
22  No internet service  No internet service  No internet service   
27                  Yes                   No                   No   
33  No internet service  No internet service  No internet service   

            StreamingTV      StreamingMovies        Contract PaperlessBilling  \
0                    No                   No  Month-to-month              Yes   
20                   No                  Yes  Month-to-month              Yes   
22  No internet service  No internet service  Month-to-month               No   
27                   No                   No  Month-to-month               No   
33  No internet service  No internet service  Month-to-month               No   

                PaymentMethod  MonthlyCharges  TotalCharges Churn  
0            Electronic check           29.85         29.85    No  
20           Electronic check           39.65         39.65   Yes  
22               Mailed check           20.15         20.15   Yes  
27           Electronic check           30.20         30.20   Yes  
33  Bank transfer (automatic)           20.20         20.20    No

Нашей гипотезе есть место быть.

Но давайте проверим:

In [56]:
print('Количество совпадений по нашей теории:')
print(df[(df['TotalCharges'] / df['MonthlyCharges']) == df['tenure']]['customerID'].count())
print('Количество различий:')
print(df[(df['TotalCharges'] / df['MonthlyCharges']) != df['tenure']]['customerID'].count())

Количество совпадений по нашей теории:
614
Количество различий:
6429


**Как видно**, что различий очень много, относительно одинаковых значений.

Но, а если мы проверим наше различие в долях или процентах? Проверяем:

In [57]:
# cоздаем series и считаем TotalCharges вручную
new_total = df['MonthlyCharges'] * df['tenure']

# посчитаем во сколько раз в среднем отличается наш new_total и TotalCharges - в процентном отношении
perc = abs(100 - (new_total / df['TotalCharges']).mean() * 100)

print(f'В среднем new_total отличается от TotalCharges на {perc:.4f}%')

В среднем new_total отличается от TotalCharges на 0.2311%


Как видно, они различаются незначительно (меньше `1%`), от чего мы можем предположить, что `TotalCharges` = `MonthlyCharges` * `время`. Скорее всего такие маленькие различия связаны с тем, что `tenure` округлялся.

В данном датасете указано только количество месяцев, без уточнения времени (до дней), потому будет грубо просто перемножить один столбец на другой. Но пропуски мы уже можем заменить на `0`, так как в тех строках `tenure` == 0. Такое небольшое исследование было сделано для того, чтобы заполнить наши пропуски нужным числом.

Заменим пропуски на нужное нам число.

In [58]:
df.fillna({'TotalCharges':0}, inplace=True)

### Изменение типа данных

Рассмотрев все данные, замените типы на нужные (при необходимости):

- Если есть числа - на `int` или `float`
- Если категории - можно оставить `object`

In [59]:
# ваше решение
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


**Вывод:**
Считаю, что данные типы данных можно оставить

# 4 Проверка гипотез

Задание: сравнить две выборки - `Churn='No'` и `Churn='Yes'` 

Гипотеза - среднее количество месяцев, в течение которых клиент оставался в компании, отличается между двумя выборками.

In [60]:
# ваше решение
df_no = df.loc[df.Churn == 'No']['tenure']
df_no.head().T

0     1
1    34
3    45
6    22
7    10
Name: tenure, dtype: int64

In [61]:
df_yes = df.loc[df.Churn == 'Yes']['tenure']
df_yes.head().T

2      2
4      2
5      8
8     28
13    49
Name: tenure, dtype: int64

In [62]:
# H0 - среднее количество месяцев, в течение которых клиент оставался в компании, не отличается между двумя выборками.
# H1 - среднее количество месяцев, в течение которых клиент оставался в компании, отличается между двумя выборками.

_, pvalue = stats.ttest_ind(df_no, df_yes)

alpha = 0.05

if pvalue < alpha:
    print(f'P-value:{pvalue:.5f}. Отвергаем нулевую гипотезу: средние различаются.')
else:
    print(f'P-value:{pvalue:.5f}. Не отвергаем нулевую гипотезу: средние не различаются.')

P-value:0.00000. Отвергаем нулевую гипотезу: средние различаются.


**Вывод:**

Cуществует статистически значимое различие в среднем количестве месяцев, проведенных клиентами в компании, в зависимости от каких-то факторов, которые разделяют выборки на две группы.

Таким образом, можно предположить, что некий фактор влияет на то, сколько клиентов остаются в компании, и необходимо дальше исследовать этот фактор для понимания его влияния и принятия соответствующих решений.

## 5 Статистический анализ на норму

Задание: выберите числовые признаки и проведите анализ - соответствуют ли распредения нормальному?

In [63]:
# ваше решение

alpha = .05

# Определяю выборку столбцов с числовыми признаками:
numeric_cols = ['tenure', 'MonthlyCharges', 'TotalCharges']
df.loc[:, numeric_cols]

tenure  MonthlyCharges  TotalCharges
0          1           29.85         29.85
1         34           56.95       1889.50
2          2           53.85        108.15
3         45           42.30       1840.75
4          2           70.70        151.65
...      ...             ...           ...
7038      24           84.80       1990.50
7039      72          103.20       7362.90
7040      11           29.60        346.45
7041       4           74.40        306.60
7042      66          105.65       6844.50

[7043 rows x 3 columns]

**Гипотезы:**
- H0: выборка имеет распределение по Гауссу (нормальное распределение).
- H1: выборка не имеет распределения по Гауссу (не нормальное распределение).

**1. Тест Шапиро-Уилка:**
не выполняем, т.к. размер выборки > 5000 (7043) и p-value может быть неточным

In [64]:
# 2. К'2-тест Д'Агостино

for col in numeric_cols:
    _, pvalue = stats.normaltest(df[col])

    if pvalue < alpha:
        print(f'P-value:{pvalue:.5f}. Нулевая гипотеза отвергается')
    else:
        print(f'P-value:{pvalue:.5f}. Подтверждается нулевая гипотеза')

P-value:0.00000. Нулевая гипотеза отвергается
P-value:0.00000. Нулевая гипотеза отвергается
P-value:0.00000. Нулевая гипотеза отвергается


In [65]:
# 3. Критерий Андерсона-Даринга


for col in numeric_cols:
    result = stats.anderson(df[col])

    for i in range(len(result.critical_values)):
        sl, cv = result.significance_level[i], result.critical_values[i]
        if result.statistic < result.critical_values[i]:
            print(f'Уровень значимости {sl:.2f}% : критическое значение {cv:.3f}, данные выглядят нормально (не удается отклонить H0)')
        else:
            print(f'Уровень значимости {sl:.2f}% : критическое значение {cv:.3f}, данные не выглядят нормально (отклоняем H0)')

Уровень значимости 15.00% : критическое значение 0.576, данные не выглядят нормально (отклоняем H0)
Уровень значимости 10.00% : критическое значение 0.656, данные не выглядят нормально (отклоняем H0)
Уровень значимости 5.00% : критическое значение 0.787, данные не выглядят нормально (отклоняем H0)
Уровень значимости 2.50% : критическое значение 0.917, данные не выглядят нормально (отклоняем H0)
Уровень значимости 1.00% : критическое значение 1.091, данные не выглядят нормально (отклоняем H0)
Уровень значимости 15.00% : критическое значение 0.576, данные не выглядят нормально (отклоняем H0)
Уровень значимости 10.00% : критическое значение 0.656, данные не выглядят нормально (отклоняем H0)
Уровень значимости 5.00% : критическое значение 0.787, данные не выглядят нормально (отклоняем H0)
Уровень значимости 2.50% : критическое значение 0.917, данные не выглядят нормально (отклоняем H0)
Уровень значимости 1.00% : критическое значение 1.091, данные не выглядят нормально (отклоняем H0)
Уровен

**Вывод:**

Во всех трех статистических тестах на норму H0 отклоняется, т.е. распределение выборки не является нормальным